server/src/main/java/io/crate/execution/engine/collect/files/FileInputFactory.java

```diff
@@ -24,8 +24,15 @@
 
 import java.io.IOException;
 import java.net.URI;
+import java.util.Map;
 
-public interface FileInputFactory {
+public abstract class FileInputFactory {
 
-    FileInput create(URI uri) throws IOException;
+    protected abstract FileInput doCreate(URI uri, Map<String, Object> withClauseOptions) throws IOException;
+
+    FileInput create(URI uri, Map<String, Object> withClauseOptions) throws IOException {
+        return doCreate(uri, validate(withClauseOptions));
+    }
+
+    protected abstract Map<String, Object> validate(Map<String, Object> withClauseOptions) throws IllegalArgumentException;
```

Comment:
Could we change this to have the `FileInputFactory` return optional and required options? Maybe even utilize the `Setting` class for the definition. That would allow each individual option to have their own validator.

And we can make the core validation part of the server framework - and make it unnecessary for each `FileInputFactory`  to implement it.